### PR TITLE
changed readme

### PR DESCRIPTION
**Pull request by Asingjr2**
please accept

### CODE REVIEW

File: copy_cat/settings/common.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
     # External
     'storages',
     # Local
-    'forum'
+    'forum.apps.ForumConfig'
```
Comment (loftsmartmatt) [inline copy_cat/settings/common.py:21]: Why did you change this? You shouldn't need to.

File: forum/templates/forum/log_reg.html [hunk 2]
```diff
@@ -7,6 +7,23 @@
...
+    <h3>Login Form</h3>
+    <form action="log_reg" method="post">
+        {% csrf_token %}
+        <input type="hidden" name="form-type" value="login">
+        {{ form2.as_p }}
+        <input type="submit" value="LOGIN USER">
+    </form>
 </body>
 </html>
 <!-- Page will exist for functionality and then I would add more complicated jquery to allow for separate iframe/div to appear for what could be ajax information flow
```
Comment (loftsmartmatt) [inline forum/templates/forum/log_reg.html]: See that at the bottom? It's saying the file is missing a newline at the very end. It's considered a style best practice to have one.

File: forum/urls.py [hunk 3]
```diff
@@ -1,16 +1,12 @@
...
 app_name = "forum"
 urlpatterns = [
-    # urls shared by everyone with index == login
-    url(r'^$', views.views, name="index"), 
-    url(r'^post$', views.post, name="post"), 
-    url(r'^logout$', views.logout, name="logout"), 
+    # Log_Reg page removed 
+    path("", views.HomePageView.as_view(), name="home"), 
+    path("user", views.UserView.as_view(), name="user"),
+    path("comment/<int:pk>", views.CommentDetailView.as_view(), name="comment_detail"),
```
Comment (loftsmartmatt) [inline forum/urls.py]: `<int:pk>` isn't going to work. That will only work if the model primary key is an integer. If you check the `Comment`, you'll notice we're using a `UUIDField` as the primary key. You'll need to change this to something else.

File: forum/views.py [hunk 4]
```diff
@@ -1,48 +1,69 @@
...
+        vote_count = Post.objects.first().votes.count()
+        upvotes = Post.objects.first().votes.filter(vote=1).count()
+        downvotes = Post.objects.first().votes.filter(vote=-1).count()
+        total = upvotes - downvotes
+        context = {
+            "vc": vote_count,
+            "total":total
+        }
+        print(context)
+        return Post.objects.all()
```
Comment (loftsmartmatt) [inline forum/views.py]: What are you trying to do here exactly? You're making 4 database calls but only returning one of them.

File: forum/views.py [hunk 5]
```diff
@@ -1,48 +1,69 @@
...
+        return redirect("/")
+
+# Detail view containing some specific user data...could evolve to user preferences page
+class UserView(DetailView):
+    model = User
+
+    def get(self, request):
+        user = request.user
+        context = {
+            "user":user,
```
Comment (loftsmartmatt) [inline forum/views.py]: FYI, you don't *need* to pass in `user` to the view. You already have access to that in all templates by calling `{{ request.user }}`.

**Review by loftsmartmatt — CHANGES_REQUESTED** [review]
Have some questions and suggestions

File: copy_cat/settings/common.py (hunk 1 above)
Comment (Asingjr2) [reply copy_cat/settings/common.py:21]: its something that was in the tutorial that i thought may be related to overall app.  Changed back and it still works!

File: forum/urls.py (hunk 3 above)
Comment (Asingjr2) [reply forum/urls.py]: Hey you are exactly right about UUID.  I didnt realize that an algorithm would be used to create an id.  Its much different than simply making an id unique.  Would like to see your thoughts on Django approach to passing UUID through url.  The first comment info could be passed as a string I imagine as i can get() the comment if i already know the id (e.g. eb0af0ce6347436690885cf3cb4566c7).  took out of url for now.

File: forum/views.py (hunk 4 above)
Comment (Asingjr2) [reply forum/views.py]: I was running test pulls to try and get total.  Thinking that I will run for loop that access each comment and input formula to total scores.  Seems unsafe so I may need address this manually OR I can update the model to create add function for total vote score [upvotes(+1) plus downvotes (-1s)].

File: forum/views.py (hunk 5 above)
Comment (Asingjr2) [reply forum/views.py]: Tried this in user but it does not seem to work for some reason.  Removed context pass.  Security element since user.is_logged_in may not be true?  Will need to test with you.
{{ request.user }} in HTML = Anonymous User
{{ user}} in HTML = Anonymous User
{{ user.first_name }} in HTML = blank
 {{ user.id }} in HTML = None

File: forum/views.py (hunk 4 above)
Comment (Asingjr2) [reply forum/views.py]: I think adding function or attribute/ property to model is good idea.  Your thoughts?

File: forum/templates/forum/log_reg.html (hunk 2 above)
Comment (Asingjr2) [reply forum/templates/forum/log_reg.html]: Added another blank line in this file and other html files.  Not sure if that is what you mean.  Can remove for now until package is added to reduce unnecessary elements.

**Comment by Asingjr2** [comment]
Added comments to MP recommendations.

File: forum/urls.py (hunk 3 above)
Comment (matthewmpalen) [reply forum/urls.py]: [You can just use `uuid`](https://docs.djangoproject.com/en/2.0/topics/http/urls/#path-converters)

File: forum/views.py (hunk 5 above)
Comment (matthewmpalen) [reply forum/views.py]: That's expected behavior. If you're looking at a page and you're not logged in, Django defaults you to `AnonymousUser`. If you try `request.user.is_authenticated`, it'll tell you it's `False`.

If you want to test the templates with an authenticated user, you should look at it while you're a superuser.

File: forum/views.py (hunk 4 above)
Comment (matthewmpalen) [reply forum/views.py]: This is not correct though. You're using `first()` which one gets a single `Post`. You already have access to `Post.objects.all()` which is a many `Post` objects. Each of those has access to `votes`, right? So I don't see why you need those extra queries since they don't really do what you want.

File: forum/views.py [hunk 6]
```diff
@@ -1,3 +1,69 @@
...
+        total = upvotes - downvotes
+        context = {
+            "vc": vote_count,
+            "total":total
+        }
+        print(context)
+        return Post.objects.all()
+
+
+def posttest(request):
```
Comment (matthewmpalen) [inline forum/views.py]: This is not a good name, and this function won't work if you're not authenticated. Wrapping the whole thing in a `try-except` block is considered bad practice because you won't know what kind of error is raised if an unhandled exception occurs.

File: forum/static/forum/jquery.js [hunk 7]
```diff
@@ -0,0 +1,4 @@
+$(document).ready(function () {
+    alert('starting reddit clone');
+    // Add in more as required
+})
```
Comment (loftsmartmatt) [inline forum/static/forum/jquery.js]: You should probably remove this file or at least comment it out. Also, `jquery.js` is not a good name.

Comment (Asingjr2) [reply forum/static/forum/jquery.js]: will comment out for now
